components/youtube_video/youtube_video.jsx

```diff
@@ -20,62 +20,16 @@ export default class YoutubeVideo extends React.PureComponent {
 
         this.state = {
             playing: false,
+            ...updateStateFromProps(props),
+            link: '',
         };
     }
 
-    UNSAFE_componentWillMount() { // eslint-disable-line camelcase
-        this.updateStateFromProps(this.props);
-    }
-
-    UNSAFE_componentWillReceiveProps(nextProps) { // eslint-disable-line camelcase
-        this.updateStateFromProps(nextProps);
-    }
-
-    updateStateFromProps = (props) => {
-        const link = props.link;
-
-        const match = link.trim().match(ytRegex);
-        if (!match || match[1].length !== 11) {
-            return;
-        }
-
-        if (props.show === false) {
-            this.stop();
+    static getDerivedStateFromProps(nextProps, prevState) {
+        if (nextProps.link !== prevState.link) {
+            return updateStateFromProps(nextProps);
```

Comment:
@hmhealey I am hoping I got you correctly on this. This is the change I made with regards to this, I have changed the if statement in `static getDerivedStateFromProps`. That looks like so now;
```
    static getDerivedStateFromProps(props, state) {
        const match = getLinkMatch(props.link);
        const matchNotNull = match !== null;
        if (state.playing && props.show && matchNotNull) {
            return updateStateFromProps(props, match);
        }
        return null;
    }
```

Reply:
I also extracted `getMatchLink` function from  the `updateStateFromProps` function.

Reply:
No, I don't think this works, unfortunately.

Now that I'm thinking more about this, I think we can almost get rid of `getDerivedStateFromProps` entirely. We don't really need to store the video ID or timestamp in `state` since it's relatively easy to calculate it.

I think we can make it look something like this:
```javascript
export default class YoutubeVideo extends React.PureComponent {
     static getDerivedStateFromProps(props, state) {
         if (!props.show && state.playing) {
             // Stop playing when the video is closed
             return {playing: false};
        }

        return null;
    }

    render() {
        const videoId = getVideoId(this.props.link); // This could also use getLinkMatch, but it would be a bit simpler if we change that function to just return the ID
        
        const time = handleYoutubeTime(this.props.link);
 
        return ...;
    }
}
```